task/source-build-oci-ta/0.1/source-build-oci-ta.yaml

```diff
@@ -52,16 +58,55 @@ spec:
         - use
         - $(params.SOURCE_ARTIFACT)=/var/workdir/source
         - $(params.CACHI2_ARTIFACT)=/var/workdir/cachi2
+    - name: get-base-images
+      image: quay.io/konflux-ci/appstudio-utils:ab6b0b8e40e440158e7288c73aff1cf83a2cc8a9@sha256:24179f0efd06c65d16868c2d7eb82573cce8e43533de6cea14fec3b7446e0b14
+      env:
+        - name: BASE_IMAGES
+          value: $(params.BASE_IMAGES)
+      script: |
+        #!/usr/bin/env bash
+        set -euo pipefail
+
+        if [[ -n "$BASE_IMAGES" ]]; then
+            echo "BASE_IMAGES param received:"
+            printf "%s" "$BASE_IMAGES" | tee "$BASE_IMAGES_FILE"
+            exit
+        fi
+
+        echo "BASE_IMAGES param is empty, inspecting the SBOM instead"
+
+        raw_inspect=$(skopeo inspect --raw "docker://$BINARY_IMAGE")
+        if manifest_digest=$(jq -e -r '.manifests[0].digest' <<<"$raw_inspect"); then
+            # The BINARY_IMAGE is an index image, each manifest in the list has its own SBOM.
+            # We're gonna assume the base images are the same or similar enough in all the SBOMs.
+            # How would we even handle a build where each manifest in the list is built from different base images?
+            echo "BINARY_IMAGE ($BINARY_IMAGE) is a manifest list, picking an arbitrary image from the list"
+            image_without_digest=${BINARY_IMAGE%@*}
+            image_without_tag=${image_without_digest%:*}
+            image=${image_without_tag}@${manifest_digest}
+        else
+            # The image is a single manifest
+            image=$BINARY_IMAGE
+        fi
+
+        echo "Downloading SBOM for $image"
+        sbom=$(cosign download sbom "$image")
```

Comment:
Should we gracefully handle errors? Do we need a retry mechanism?

Reply:
:+1: added basic retry mechanism. Not sure if we should let the source build proceed when we fail to download the SBOM. It is true that the source build task is generally very tolerant to failing to find base image source. WDYT @tkdchen ?

Reply:
Yeah. Any failure during handling the base image sources should not break the whole build. I would prefer keeping the graceful handle with meaningful logs.

Reply:
Changed it to proceed even if the SBOM can't be downloaded